features/step_definitions/operators.rb

```diff
@@ -160,3 +160,35 @@
   cb_name = 'operator_channel_name' unless cb_name
   cb[cb_name] = cluster_version('version').channel.split('-')[1]
 end
+
+Given /^I wait(?: up to (\d+) seconds)? for operator "([^"]*)" to check conditions: "([^"]*)"$/ do | time, operator, conditions |
+  ensure_admin_tagged
+
+  interval_time = 15
+  expected = {}
+  # Parse the conditions to Hash value {Available"=>"True", "Progressing"=>"False", "Degraded"=>"False", "Upgradeable"=>"True"}
+  if conditions.include? "="
+    conditions.split(/\s+/).map{|ar| ar.split('=')}.each {|k, v| expected[k]=v}
+  else
+    arr_conditions = conditions.split(/\s+/)
+    ["Available", "Progressing", "Degraded", "Upgradeable"][0,arr_conditions.length].zip(arr_conditions).map {|k, v| expected[k]=v}
+  end
+  raise "### Without valid conditions, unable to continue to check !!!" unless expected.length >0
+
+  time = Integer(time) rescue 60
+  interval_time = 30 if time > 100
```

Comment:
30 is too coarse. If using 30, then 100s only tries at 30s, 60s, 90s. If success at 40s, then it still needs wait 20s to exit the loop.
I think need not use interval_time

Reply:
Without interval_time,  waiting time will be the whole timeout value, the more time to wait is uncertain. This is not an optimal solution. With interval_time, the more time to wait is interval_time in the worst case, will quit the process early.

Reply:
For example,
```
timeout = 100 
with interval = 10
10,20,30,40,50,60,70,80,90,100 <- attempts
best case: more time to wait -> 0
worst case: more time to wait -> 10

without interval,
one attempt
we assume the program quit in 10s, 50s, 100s
more time to wait -> 90s, 50s, 0s.

So with internal, the result is better.
```

Reply:
No. Check lib/base_helper.rb:143:      def wait_for(seconds, interval: 1, stats: nil)
Default interval is 1 if not set.

Reply:
yes, I noticed this, interval:1 is inadvisable, more attempts. An appropriate increase in interval time is desirable for a long wait.

Reply:
OK, you can change `> 100` to larger:
  interval_time = 30 if time > 600

Reply:
Attempt once a second is too frequent, it is necessary to extend the interval time for detecting the status of operator. I adjust the interval time as below, 
```
  interval_time = 5
  ...
  interval_time = 20 if time > 100
```

Reply:
> OK, you can change > 100 to larger:
> interval_time = 30 if time > 600

This comment already agreed to your previous comment. OK, you can feel free to adjust the values.

Reply:
There is something wrong with page refreshing, sometime I couldn't get the new content.